framework/TApplication.php

```diff
@@ -1104,6 +1108,20 @@ public function onError($param)
 		$this->getErrorHandler()->handleError($this, $param);
 	}
 
+	/**
+	 * Raises onInitComplete event.
+	 * At the time when this method is invoked, application modules are loaded,
+	 * user request is resolved and the corresponding service is loaded and
+	 * initialized. The application is about to start processing the user
```

Comment:
At this step services are loaded  but not initialized yet: they have no ID assigned and their still have to load their config.